apps/kg/dataloader/KGDataset.py

```diff
@@ -150,27 +291,201 @@ def read_triple(self, path, mode, skip_first_line=False):
         print('Finished. Read {} {} triples.'.format(len(heads), mode))
         return (heads, rels, tails)
 
+class KGDatasetUDDRaw(KGDataset):
+    '''Load a knowledge graph user defined dataset
+
+    The user defined dataset has five files:
+    * entities stores the mapping between entity name and entity Id.
+    * relations stores the mapping between relation name relation Id.
+    * train stores the triples in the training set. In format [src_name, rel_name, dst_name]
+    * valid stores the triples in the validation set. In format [src_name, rel_name, dst_name]
+    * test stores the triples in the test set. In format [src_name, rel_name, dst_name]
+
+    The mapping between entity (relation) name and entity (relation) Id is stored as 'name\tid'.
+    The triples are stored as 'head_nid\trelation_id\ttail_nid'.
+    '''
+    def __init__(self, path, name, files, format):
+        self.name = name
+        for f in files:
+            assert os.path.exists(os.path.join(path, f)), \
+                'File {} now exist in {}'.format(f, path)
+
+        assert len(format) == 3
+        format = _parse_srd_format(format)
+        self.load_entity_relation(path, files, format)
+
+        if len(files) == 1:
```

Comment:
I suppose it means the training data is provided not not validation data and test data.